rasa/core/policies/rule_policy.py

```diff
@@ -224,6 +244,23 @@ def _create_loop_unhappy_lookup_from_states(
                 lookup[feature_key] = DO_NOT_PREDICT_LOOP_ACTION
         return lookup
 
+    def _check_rule_trackers(self, rule_trackers):
+        bad_rules = []
+        for tracker in rule_trackers:
+            number_of_user_uttered = sum(
+                isinstance(event, UserUttered) for event in tracker.events
+            )
+            if number_of_user_uttered > self.ALLOWED_NUMBER_OF_USER_INPUTS:
+                bad_rules.append(tracker.sender_id)
+
+        if bad_rules:
+            raise InvalidRule(
+                f"Found rules {bad_rules} that contain more than "
+                f"{self.ALLOWED_NUMBER_OF_USER_INPUTS} user inputs. "
+                f"Rules are not meant to hardcode a state machine. "
+                f"Please use stories for these cases."
```

Comment:
let's include the link to the rule documentation. there is a constant in `rasa.constants` (DOCS_URL_RULES)